sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -555,4 +558,37 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
 
     assert(numbers.count() == 8)
   }
+
+  test("load data with empty quoted string fields.") {
+    val results = sqlContext
+      .read
+      .format("csv")
+      .options(Map(
+        "header" -> "true",
+        "nullValue" -> null,
```

Comment:
If nullValue is not set  it  will return empty string for null values by default.  The reason I set it explicitly is to make sure my fix is working.  Before my fix it was  retruning null for the empty quoted string , and empty  string for null values by default.